### ISSUE TITLE
[4.x]: Shipping zone can't have the same name as a tax zone

### ISSUE THREAD

**Issue by TomFaust** [craftcms/commerce#3317]
### Description
While trying to save a shipping zone I received the error message saying the name  'Canada' was already in use, however I had no other shipping zones setup. Looking in the database I saw no records in the shipping zones tables either, nor anything named 'Canada' in the shipping methods or shipping category sections. Eventually I noticed that i had a tax zone named 'Canada', and when changing this to 'Canadab' I was able to save my shipping zone. However, going back to the tax zone I was still able to save it as 'Canada' no problem.

### Steps to reproduce
1. Create a tax zone called 'Canada'
2. Attempt to create a shipping zone called 'Canada'

### Expected behavior
Save the shipping zone

### Actual behavior
Unable to save, seemingly because there is a tax zone present with the same name

### Craft CMS version

4.5.3

### Craft Commerce version

4.2.11

### PHP version

8.1

### Operating system and version

macOS 13.3 (22E252)

### Installed plugins and versions

-

**Comment by lukeholder** [comment]
Thanks for reporting this! I have fixed it for the next release.

To get the fix early, change your `craftcms/commerce` requirement in composer.json to:

```json
"require": {
  "craftcms/commerce": "dev-develop#6e79084b5ca6adc660f2a6d53d2a76947316b4ce as 4.3.1",
  "...": "..."
}
```

Then run `composer update`.

I will also update this issue once the release is out.

**Comment by lukeholder** [comment]
(Fixed in 6e79084b5ca6adc660f2a6d53d2a76947316b4ce)

**Comment by lukeholder** [comment]
This is now fixed in 4.3.2
